src/commands/project/retrieve/start.ts

```diff
@@ -222,9 +223,15 @@ export default class RetrieveMetadata extends SfCommand<RetrieveResultJson> {
         })
       : new RetrieveResultFormatter(this.retrieveResult, flags['package-name'], fileResponsesFromDelete);
     if (!this.jsonEnabled()) {
-      // in the case where we didn't retrieve anything, check if we have any deletes
       if (this.retrieveResult.response.status === 'Succeeded' || fileResponsesFromDelete.length !== 0) {
         await formatter.display();
+      } // in the case where we didn't retrieve anything, check if we have any deletes
+      else if (
+        componentSetFromNonDeletes.size === 0 &&
+        fileResponsesFromDelete.length === 0 &&
+        (flags.manifest || flags.metadata || flags['source-dir'])
+      ) {
+        throw new SfError('No metadata retrieved');
```

Comment:
worth a unit test or nut?

Reply:
Steve, yes, will do to go with the STL PR.
Willie...trying to make it match how deploy works (retrieve "changes" can be empty, but retrieves with specified material should not).  ref: https://github.com/forcedotcom/cli/discussions/2065

Reply:
FyI, the retrieve will retrieve an "empty" package.xml so even when nothing retrieves, it still looks like a success.  We'd have to do more work to make it NOT do that (ex: check that there's anything in there besides package.xml) so I'm inclined to leave it as a success (the formatter will give a warning since the user *probably* didn't intend that).

The "there was nothing in the CS for a pull-like retrieve" output also moved to the formatter.  It seems to work well.

There's also a NUT that helps make sure that an empty pull-like retrieve is a exit0.